subsys/bluetooth/mesh/net.c

```diff
@@ -1329,7 +1324,7 @@ void bt_mesh_net_recv(struct net_buf_simple *data, s8_t rssi,
 	 */
 	if (bt_mesh_trans_recv(&buf, &rx) == -EAGAIN) {
 		BT_WARN("Removing rejected message from Network Message Cache");
-		msg_cache[rx.msg_cache_idx] = 0ULL;
+		msg_cache[rx.msg_cache_idx].src = BT_MESH_ADDR_UNASSIGNED;
```

Comment:
Would it be cleaner to add a msg_cache_remove for this?

Reply:
We can't really remove any entry from the cache though, so I'm afraid a generic sounding remove function will just open up for a potential future bug. I can implement my other comment to block this seemingly good idea in the future though.